components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -842,18 +849,15 @@ private void createTenant(String domain, String orgCreatorID, String orgCreatorN
         }
     }
 
-    private Tenant createTenantInfoBean(String domain, String orgCreatorID) {
+    private Tenant createTenantInfoBean(String domain, String orgCreatorID, String orgCreatorName,
+                                        String orgCreatorEmail) {
 
         Tenant tenant = new Tenant();
         tenant.setActive(true);
         tenant.setDomain(domain);
-        /*
-        Set the creator's UUID as the tenant admin name because tenant data model doesn't store the admin uuid,
-        and it is required when creating the org-user association.
-         */
-        tenant.setAdminName(orgCreatorID);
+        tenant.setAdminName(orgCreatorName);
         tenant.setAdminUserId(orgCreatorID);
-        tenant.setEmail("dummyadmin@email.com");
+        tenant.setEmail(orgCreatorEmail);
         // set the password as domain for now to avoid findbugs detecting it as a hardcoded value.
         tenant.setAdminPassword(domain);
```

Comment:
I think we can remove this line.

Reply:
rebased the code. removed the password setting from a previous PR